test/extended/util/framework.go

```diff
@@ -984,13 +984,20 @@ func WaitForServiceAccount(c corev1client.ServiceAccountInterface, name string)
 			return false, fmt.Errorf("Failed to get service account %q: %v", name, err)
 		}
 		secretNames := []string{}
+		var hasDockercfg, hasToken bool
 		for _, s := range sc.Secrets {
-			if strings.Contains(s.Name, "dockercfg") {
-				return true, nil
+			if strings.Contains(s.Name, "-dockercfg-") {
```

Comment:
This one needs to be in `sc.ImagePullSecrets`.

Reply:
Good catch